lib/src/base-styles/_nx-btn.scss

```diff
@@ -6,48 +6,75 @@
  */
 @use '../scss-shared/nx-container-helpers';
 @use '../scss-shared/nx-text-helpers';
+@use '../scss-shared/nx-dark-mode-helpers';
 
-%button-focus-outline {
+%transparent-button-focus-outline {
+  outline-offset: calc(0px - var(--nx-border-width-button) - 4px);
+  outline: 2px solid;
+}
+
+%solid-button-focus-outline {
   outline-offset: -5px;
   outline: 2px solid;
 }
 
+@mixin solid-btn-styling($variant) {
+  background-color: var(--nx-color-#{$variant}-button-background);
+  border-color: transparent;
+  color: var(--nx-color-#{$variant}-button-text);
+
+  &:hover {
+    background-color: var(--nx-color-#{$variant}-button-background-hover);
+  }
+
+  &:focus:not(:active) {
+    @extend %solid-button-focus-outline;
+    outline-color: var(--nx-color-#{$variant}-button-focus-outline);
+  }
+
+  &:active {
+    background-color:  var(--nx-color-#{$variant}-button-background-active);
+    border-color: transparent;
+  }
+}
+
 .nx-btn {
   @include nx-text-helpers.semi-bold;
   @include nx-container-helpers.container-horizontal;
 
-  background-color: var(--nx-swatch-white);
-  border: 1px solid var(--nx-swatch-blue-40);
+  background-color: var(--nx-color-secondary-button-background);
+  border: var(--nx-border-width-button) solid var(--nx-color-secondary-button-border);
   border-radius: var(--nx-border-radius);
-  color: var(--nx-swatch-blue-40);
+  color: var(--nx-color-secondary-button-text);
   cursor: pointer;
   display: inline-block;
   font-size: var(--nx-font-size);
   line-height: var(--nx-line-height);
   margin: 0 var(--nx-spacing-6x);
-  padding: calc(var(--nx-spacing-2x) - 1px) calc(var(--nx-spacing-6x) - 1px);
+  padding: calc(var(--nx-spacing-2x) - var(--nx-border-width-button)) calc(var(--nx-spacing-6x) - var(--nx-border-width-button));
   text-align: center;
   text-decoration: none;
 
   &:hover {
-    background-color: var(--nx-swatch-blue-90);
+    background-color: var(--nx-color-secondary-button-background-hover);
   }
 
   &.nx-btn--secondary:hover {
-    color: var(--nx-swatch-blue-30);
+    color: var(--nx-color-secondary-button-text-hover);
```

Comment:
Why is this in a separate block and not the `&:hover` block above? Note that we do still support raw HTML buttons which use the `nx-btn` classes directly, and in that support we also do still say that a default `nx-btn` is a secondary `nx-btn`.  That is, `<button class="nx-btn">` and `<button class="nx-btn nx-btn--secondary">` should behave identically.  I suspect the only reason this isn't causing a bug is because `--nx-color-secondary-button-text-hover` is the same as `--nx-color-secondary-button-text` (what the button has without this line) anyway.

Reply:
> Why is this in a separate block and not the `&:hover` block above? Note that we do still support raw HTML buttons which use the `nx-btn` classes directly, and in that support we also do still say that a default `nx-btn` is a secondary `nx-btn`. That is, `<button class="nx-btn">` and `<button class="nx-btn nx-btn--secondary">` should behave identically. I suspect the only reason this isn't causing a bug is because `--nx-color-secondary-button-text-hover` is the same as `--nx-color-secondary-button-text` (what the button has without this line) anyway.

This had stemmed from a problem that arose in the interaction state button ticket. For the secondary button, when hovered, the text color shifts slightly from `blue-40` to `blue-30` to pass contrast. By adding the color change in the `&:hover` block, it also affected the text color on hover of all other buttons. Would you suggest listing it inside of the `&:hover` block and then override it in the other hover states?

Reply:
> Would you suggest listing it inside of the &:hover block and then override it in the other hover states?

Yes

Reply:
> > Would you suggest listing it inside of the &:hover block and then override it in the other hover states?
>
> Yes

To be clear, after your explanation I can see that there is already a bug here in light mode: on raw HTML buttons with the `nx-btn` class, the hover text color is the same as the non-hover text color when it should be 10% lighter.

Reply:
> > > Would you suggest listing it inside of the &:hover block and then override it in the other hover states?
> > 
> > 
> > Yes
> 
> To be clear, after your explanation I can see that there is already a bug here in light mode: on raw HTML buttons with the `nx-btn` class, the hover text color is the same as the non-hover text color when it should be 10% lighter.

Updated here: [db199c1](https://github.com/sonatype/sonatype-react-shared-components/pull/1038/commits/db199c1c73ec523021856baca2e132fef8235309)